whiteout/resolver_test.go

```diff
@@ -0,0 +1,292 @@
+					},
+				},
+				Environments: map[string][]*claircore.Environment{
+					"1": {{IntroducedIn: firstLayerHash}},
+					"2": {{IntroducedIn: firstLayerHash}},
+				},
+				Files: map[string]claircore.File{
+					firstLayerHash.String(): { // whiteout is in the same layer as packages
+						Path: "a/path/to/some/file/site-packages/.wh.b_package",
+						Kind: claircore.FileKindWhiteout,
+					},
+				},
+			},
+			layers: []*claircore.Layer{
+				{Hash: firstLayerHash},
+				{Hash: secondLayerHash},
+			},
+			lenPackage: 2,
+			lenEnvs:    2,
+		},
+		{
+			name: "common prefix, distinct dirs test",
+			ir: &claircore.IndexReport{
+				Packages: map[string]*claircore.Package{
+					"1": {
+						Name:     "something interesting",
+						Filepath: "a/path/to/some/file/site-packages/a_package/METADATA",
+					},
+					"2": {
+						Name:     "something uninteresting",
+						Filepath: "a/path/to/some/file/site-packages/b_package/METADATA",
+					},
+				},
+				Environments: map[string][]*claircore.Environment{
+					"1": {{IntroducedIn: firstLayerHash}},
+					"2": {{IntroducedIn: firstLayerHash}},
+				},
+				Files: map[string]claircore.File{
+					secondLayerHash.String(): {
+						Path: "a/path/to/some/file/site/.wh..wh..opq",
+						Kind: claircore.FileKindWhiteout,
+					},
+				},
+			},
+			layers: []*claircore.Layer{
+				{Hash: firstLayerHash},
+				{Hash: secondLayerHash},
+			},
+			lenPackage: 2,
+			lenEnvs:    2,
+		},
+		{
+			name: "opaque test",
+			ir: &claircore.IndexReport{
+				Packages: map[string]*claircore.Package{
+					"1": {
+						Name:     "something interesting",
+						Filepath: "a/path/to/some/file/site-packages/a_package/METADATA",
+					},
+					"2": {
+						Name:     "something uninteresting",
+						Filepath: "a/path/to/some/file/site-packages/b_package/METADATA",
+					},
+				},
+				Environments: map[string][]*claircore.Environment{
+					"1": {{IntroducedIn: firstLayerHash}},
+					"2": {{IntroducedIn: firstLayerHash}},
+				},
+				Files: map[string]claircore.File{
+					secondLayerHash.String(): {
+						Path: "a/path/to/some/file/site-packages/.wh..wh..opq",
+						Kind: claircore.FileKindWhiteout,
+					},
+				},
+			},
+			layers: []*claircore.Layer{
+				{Hash: firstLayerHash},
+				{Hash: secondLayerHash},
+			},
+			lenPackage: 0,
+			lenEnvs:    0,
+		},
+	}
+
+	r := &Resolver{}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			report := r.Resolve(context.Background(), tc.ir, tc.layers)
+			if tc.lenPackage != len(report.Packages) {
+				t.Fatalf("wrong number of packages: expected: %d got: %d", tc.lenPackage, len(report.Packages))
+			}
+			if tc.lenEnvs != len(report.Environments) {
+				t.Fatalf("wrong number of environments: expected: %d got: %d", tc.lenEnvs, len(report.Environments))
+			}
+		})
+
+	}
+}
+
+func TestIsDeleted(t *testing.T) {
+	type testcase struct {
+		Path     string
+		Whiteout string
+		Deleted  bool
+	}
+
+	for _, tc := range []testcase{
```

Comment:
can you add
```
{
	Path:     "a",
	Whiteout: "a/.wh..wh..opq",
	Deleted:  false,
},
```

I believe this should be `false` but I think we'd get `true` based on the implementation? Let me know your thoughts on this